js/apps/admin-ui/src/clients/authorization/ResourcesPolicySelect.tsx

```diff
@@ -108,57 +128,133 @@ export const ResourcesPolicySelect = ({
     [search]
   );
 
+  const [toggleUnsavedChangesDialog, UnsavedChangesConfirm] = useConfirmDialog({
+    titleKey: t("common:unsavedChangesTitle"),
+    messageKey: t("common:unsavedChangesConfirm"),
+    continueButtonLabel: t("common:continue"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: async () => {
+      navigate(
+        toPolicyDetails({
+          realm: realm,
+          id: clientId,
+          policyId: clickedPolicy?.id!,
+          policyType: clickedPolicy?.type!,
+        })
+      );
+    },
+  });
+
   const toSelectOptions = () =>
     items.map((p) => (
       <SelectOption key={p.id} value={p.id}>
         {p.name}
       </SelectOption>
     ));
 
+  const toChipGroupItems = (
+    field: ControllerRenderProps<PolicyRepresentation, Type>
+  ) => {
+    return (
+      <ChipGroup>
+        {(field.value as string[]).map((permissionId) => {
```

Comment:
Rather than type-casting this value, I'd prefer if we properly handle the possibility this value might be undefined. Trying to out-smart the compiler can lead to undefined behavior. 
```suggestion
        {field.value?.map((permissionId) => {
```